src/test/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumerTest.java

```diff
@@ -25,96 +31,161 @@
+        // Then
+        verify(ordersKafkaProducer, times(1)).sendMessage(any());
+    }
+
+    @Test
+    public void republishMessageToErrorTopicRunsSuccessfully()
+            throws ExecutionException, InterruptedException, SerializationException {
         // Given & When
-        ordersKafkaConsumer.setChKafkaConsumerGroupMain(chKafkaConsumerGroupMain);
         when(serializerFactory.getGenericRecordSerializer(OrderReceived.class)).thenReturn(serializer);
         when(serializer.toBinary(any())).thenReturn(new byte[4]);
-        doCallRealMethod().when(ordersKafkaConsumer).republishMessageToTopic(anyString(), anyString(), anyString(), any());
-        ordersKafkaConsumer.republishMessageToTopic(ORDER_RECEIVED_URI, ORDER_RECEIVED_TOPIC, ORDER_RECEIVED_TOPIC_RETRY, new OrderProcessingException());
+        doCallRealMethod().when(ordersKafkaConsumer).republishMessageToTopic(anyString(), anyString(), anyString());
+        ordersKafkaConsumer.republishMessageToTopic(ORDER_RECEIVED_URI, ORDER_RECEIVED_TOPIC_RETRY, ORDER_RECEIVED_TOPIC_ERROR);
         // Then
-        verify(chKafkaConsumerGroupMain, times(3)).retry(anyInt(), any());
+        verify(ordersKafkaProducer, times(1)).sendMessage(any());
     }
 
     @Test
-    public void republishMessageToErrorTopicRunsSuccessfully() throws ExecutionException, InterruptedException, SerializationException {
+    public void republishMessageSuccessfullyCalledForFirstMainMessageOnRetryableErrorException()
+            throws SerializationException {
         // Given & When
-        ordersKafkaConsumer.setChKafkaConsumerGroupRetry(chKafkaConsumerGroupRetry);
         when(serializerFactory.getGenericRecordSerializer(OrderReceived.class)).thenReturn(serializer);
         when(serializer.toBinary(any())).thenReturn(new byte[4]);
-        doCallRealMethod().when(ordersKafkaConsumer).republishMessageToTopic(anyString(), anyString(), anyString(), any());
-        ordersKafkaConsumer.republishMessageToTopic(ORDER_RECEIVED_URI, ORDER_RECEIVED_TOPIC_RETRY, ORDER_RECEIVED_TOPIC_ERROR, new OrderProcessingException());
+        doThrow(new RetryableErrorException(PROCESSING_ERROR_MESSAGE)).when(ordersKafkaConsumer).logMessageReceived(any());
+        ordersKafkaConsumer.handleMessage(createTestMessage(ORDER_RECEIVED_TOPIC));
+        // Then
+        verify(ordersKafkaConsumer, times(1)).republishMessageToTopic(anyString(), anyString(), anyString());
+    }
+
+    @Test
+    public void republishMessageNotCalledForFirstRetryMessageOnRetryableErrorException() {
+        // Given & When
+        doThrow(new RetryableErrorException(PROCESSING_ERROR_MESSAGE)).when(ordersKafkaConsumer).logMessageReceived(any());
+        ordersKafkaConsumer.handleMessage(createTestMessage(ORDER_RECEIVED_TOPIC_RETRY));
         // Then
-        verify(chKafkaConsumerGroupRetry, times(3)).retry(anyInt(), any());
+        verify(ordersKafkaConsumer, times(0)).republishMessageToTopic(anyString(), anyString(), anyString());
```

Comment:
Could the arguments to `republishMessageToTopic()` not be made more specific?

Reply:
This test just verifies that the `republishMessageToTopic()` method doesn't get called under specific conditions, to test the retry logic, so the parameters don't matter much.